src/run/handlers/cache.cts

```diff
@@ -165,6 +166,30 @@ export class NetlifyCacheHandler implements CacheHandlerForMultipleVersions {
     }
   }
 
+  private async getPrerenderManifest(serverDistDir: string): Promise<PrerenderManifest> {
+    if (this.prerenderManifest) {
+      return this.prerenderManifest
+    }
+
+    const prerenderManifestPath = join(serverDistDir, '..', 'prerender-manifest.json')
+
+    try {
+      // @ts-expect-error Starting in 15.4.0-canary.10 loadManifest was relocated (https://github.com/vercel/next.js/pull/78358)
+      // eslint-disable-next-line import/no-unresolved, n/no-missing-import
+      const { loadManifest } = await import('next/dist/server/load-manifest.external.js')
+      this.prerenderManifest = Promise.resolve(
+        loadManifest(prerenderManifestPath) as PrerenderManifest,
+      )
+    } catch {
+      const { loadManifest } = await import('next/dist/server/load-manifest.js')
+      this.prerenderManifest = Promise.resolve(
+        loadManifest(prerenderManifestPath) as PrerenderManifest,
+      )
+    }
+
+    return this.prerenderManifest
+  }
+
```

Comment:
This would need to not be part of class instance.

I think this would also be overall good because `loadManifest` call would no longer happen on each call to `injectEntryToPrerenderManifest` (`loadManifest` seems to use memoization itself https://github.com/vercel/next.js/blob/c3429682aa910eb2b5ddd1e761c8ec8cfaa4bb04/packages/next/src/server/load-manifest.external.ts#L40-L43 , but we then no longer would need to rely on this if it ever change to avoid redoing same work for subsequent requests)